src/partners/0xgasless.ts

```diff
@@ -0,0 +1,250 @@
+  pluginParams: PluginParams
+): Promise<PluginResult> {
+  const { settings, apiKeys } = asStandardPluginParams(pluginParams)
+
+  if (apiKeys.apiKey == null) {
+    throw new Error('0xGasless: Missing 0xgasless API key')
+  }
+  const nowDate = new Date()
+  const now = nowDate.getTime()
+
+  let { latestIsoDate } = settings
+
+  if (latestIsoDate === EDGE_APP_START_DATE) {
+    latestIsoDate = new Date(PLUGIN_START_DATE).toISOString()
+  }
+
+  let lastCheckedTimestamp = new Date(latestIsoDate).getTime() - QUERY_LOOKBACK
+  if (lastCheckedTimestamp < 0) lastCheckedTimestamp = 0
+
+  const ssFormatTxs: StandardTx[] = []
+  let retry = 0
+
+  while (true) {
+    const startTimestamp = lastCheckedTimestamp
+    const endTimestamp = lastCheckedTimestamp + QUERY_TIME_BLOCK_MS
+
+    try {
+      let cursor: string | undefined
+
+      while (true) {
+        const urlObj = new URL(API_URL, true)
+
+        const queryParams: {
+          startTimestamp: string
+          endTimestamp: string
+          cursor?: string
+        } = {
+          // API expects seconds-based unix timestamps
+          startTimestamp: Math.floor(startTimestamp / 1000).toString(),
+          endTimestamp: Math.floor(endTimestamp / 1000).toString()
+        }
+        if (cursor != null) queryParams.cursor = cursor
+        urlObj.set('query', queryParams)
+
+        datelog(
+          `0xGasless Querying from:${new Date(
+            startTimestamp
+          ).toISOString()} to:${new Date(endTimestamp).toISOString()}`
+        )
+
+        const url = urlObj.href
+        const response = await retryFetch(url, {
+          headers: {
+            '0x-api-key': apiKeys.apiKey,
+            '0x-version': 'v2'
+          }
+        })
+        const responseJson = await response.text()
+        if (!response.ok) {
+          throw new Error(`${url} response ${response.status}: ${responseJson}`)
+        }
+        const responseBody = asGetGaslessTradesResponse(responseJson)
+
+        for (const trade of responseBody.trades) {
+          const buySymbol = trade.tokens.find(t => t.address === trade.buyToken)
+            ?.symbol
+          const sellSymbol = trade.tokens.find(
+            t => t.address === trade.sellToken
+          )?.symbol
+
+          if (buySymbol == null || sellSymbol == null) {
+            throw new Error(
+              `Could not find buy or sell symbol for trade ${trade.zid}; txid: ${trade.transactionHash}`
+            )
+          }
+
+          const {
+            isoDate: tradeIsoDate,
+            timestamp: tradeTimestamp
+          } = smartIsoDateFromTimestamp(trade.timestamp * 1000)
+
+          // If trade is 2 days or older, then it's finalized according to 0x
+          // documentation.
+          const status: Status =
+            tradeTimestamp + 2 * 24 * 60 * 60 * 1000 < now
+              ? 'complete'
+              : 'pending'
+
+          const ssTx: StandardTx = {
+            status,
+            orderId: trade.zid,
```

Comment:
This MUST match the orderId used by the 0x exchange plugin. I don't think it does.